llarp/service/endpoint.hpp

```diff
@@ -7,6 +7,7 @@
 #include <llarp/path/path.hpp>
 #include <llarp/path/pathbuilder.hpp>
 #include "address.hpp"
+#include "endpoint_state.hpp"
```

Comment:
i wonder if we could not require this header and forward declare as much as we can.

Reply:
This was required because I removed the destructor, but that meant the implicit one needs to be in the header.  I've made the destructor explicit again and put a defaulted one in the .cpp, which solves this.